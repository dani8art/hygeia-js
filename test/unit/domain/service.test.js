'use strict';
const { Service } = require('../../../src/checker/domain/service');

describe('Service', () => {

    test('Create a new service', () => {
        const service = new Service({
            name: 'Google',
            health: 'http://google.es',
            method: 'GET',
            timeout: 12
        });

        expect(service.name).toBe('Google');
        expect(service.health).toBe('http://google.es');
        expect(service.method).toBe('GET');
        expect(service.timeout).toBe(12);
    });

    test('Name is required', () => {
        expect(() => {
            const service = new Service({
                health: 'http://google.es'
            });
        }).toThrow('name property is required on type Service.');
    });

    test('Health is required', () => {
        expect(() => {
            const service = new Service({
                name: 'Google'
            });
        }).toThrow('health property is required on type Service.');
    });

    test('Timeout must be number', () => {
        expect(() => {
            const service = new Service({
                name: 'Google',
                health: 'http://google.es',
                timeout: '10000'
            });
        }).toThrow('timeout property must be a number.');
    });

    test('Default values', () => {
        const service = new Service({
            name: 'Google',
            health: 'http://google.es'
        });

        expect(service.method).toBe('GET');
        expect(service.timeout).toBe(Service.DEFAULT_TIMEOUT);
    });

    test('JSON Schema', () => {
        expect(Service.getJSONSchema()).toMatchSnapshot();
    });
});