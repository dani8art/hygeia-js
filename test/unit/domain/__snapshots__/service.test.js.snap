// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service JSON Schema 1`] = `
Object {
  "$schema": "http://json-schema.org/draft-07/schema#",
  "properties": Object {
    "health": Object {
      "type": "string",
    },
    "method": Object {
      "type": "string",
    },
    "service": Object {
      "type": "string",
    },
    "timeout": Object {
      "type": "number",
    },
  },
  "required": Array [
    "service",
    "health",
  ],
  "title": "Pet",
  "type": "object",
}
`;
